### PR TITLE
11088: upgrade to gradle 8.3

### PR DESCRIPTION
**Pull request by mkalish**
This PR updates gradle to 8.3

Test Steps:
1. Check this branch and refresh the gradle project
2. Run a bunch of different gradle tasks (i.e. quickRun, test, compile, run, fatJar)
3. Start the server and run the smoke tests

## Changes
- Update to gradle 8.3
- Clean up deprecation warnings in the build file

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- #11088 

## To Be Done
- #9819 
- #9818

### CODE REVIEW

**Comment by mkalish** [comment]
There are still some deprecation warnings but they are coming from plugins, specifically:
- `nu.studer.jooq`
- `com.github.johnrengelman.shadow`

File: prime-router/build.gradle.kts [hunk 1]
```diff
@@ -97,6 +97,8 @@ val reportsApiEndpointHost = (
 val jooqSourceDir = "build/generated-src/jooq/src/main/java"
 val jooqPackageName = "gov.cdc.prime.router.azure.db"
 
+val buildDir = project.layout.buildDirectory.asFile.get()
```
Comment (mkalish) [inline prime-router/build.gradle.kts:100]: `project.buildDir` is deprecated

File: prime-router/build.gradle.kts [hunk 2]
```diff
@@ -126,8 +128,6 @@ val compileKotlin: KotlinCompile by tasks
 val compileTestKotlin: KotlinCompile by tasks
 compileKotlin.kotlinOptions.jvmTarget = appJvmTarget
 compileKotlin.kotlinOptions.allWarningsAsErrors = true
-// if you set this to true, you will get a warning, which then gets treated as an error
-compileKotlin.kotlinOptions.useK2 = false
```
Comment (mkalish) [inline prime-router/build.gradle.kts:130]: This is deprecated now (and might not have been required).  If we did need to force the version there is a `language` version option.

File: prime-router/build.gradle.kts [hunk 3]
```diff
@@ -392,8 +392,9 @@ tasks.register<JavaExec>("primeCLI") {
 
     // Use arguments passed by another task in the project.extra["cliArgs"] property.
     doFirst {
-        if (project.extra.has("cliArgs")) {
-            args = project.extra["cliArgs"] as MutableList<String>
```
Comment (mkalish) [inline prime-router/build.gradle.kts:396]: Fixed this warning since this is an unsafe cast

File: prime-router/build.gradle.kts [hunk 4]
```diff
@@ -392,8 +392,9 @@ tasks.register<JavaExec>("primeCLI") {
...
+        val cliArgs = project.extra["cliArgs"]
+        if (project.extra.has("cliArgs") && cliArgs is List<*>) {
+            args = cliArgs.filterIsInstance(String::class.java)
         } else if (args.isNullOrEmpty()) {
```
Comment (mkalish) [inline prime-router/build.gradle.kts:398]: Guarantees that args will be `List<String>`

**Review by JFU-NAVA-PBC — APPROVED** [review]
LGTM

passed all tests local

thanks for the thorough refactoring

File: prime-router/gradlew [hunk 5]
```diff
@@ -193,6 +194,10 @@ if "$cygwin" || "$msys" ; then
     done
 fi
 
+
+# Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
+DEFAULT_JVM_OPTS='"-Xmx64m" "-Xms64m"'
```
[the remark this answers is not in the document]
Comment (mkalish) [reply prime-router/gradlew:199]: Oh maybe?  This file is auto-generated by gradle

[the remark this answers is not in the document]
Comment (jeremy-page) [reply prime-router/gradlew:199]: 64 meg is not a lot of heap...

[the remark this answers is not in the document]
Comment (mkalish) [reply prime-router/gradlew:199]: I believe [this](https://github.com/CDCgov/prime-reportstream/blob/589f7c24f02a33c44dbb65e41fa186196b67d9a1/prime-router/gradle.properties#L3) overrides it

**Pull request merged**